fix: remove inline style in announcer

YannicEl opened the pull request:
partly closes  #11747 

This PR moves the svelte announcer styles to it's own stylesheet rather than as inline CSS. It also ensures that the root component's styles are inlined if style inlining is enabled. This makes it possible to at least use `style-src` without any inlined styles.

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.
No test unless we want to create a new test app for `style-src 'self'`

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

YannicEl committed:
replace inline styles with style tag

benmccann commented:
Thanks for the PR. Could you look at the test failures? Don't worry if an unrelated one has failed due to flakiness, but there shouldn't be four test jobs failing

YannicEl commented:
Thanks for the insights. 

For now I applied a local patch to fix the issue for me. 

When I have more time I will look into how a strict `CSP` and `AMP` can work together.

teemingc committed:
Merge branch 'main' into unsafe_content_security_policy_fix

teemingc committed:
account for root component's stylesheets

teemingc committed:
don't add root stylesheet for inline builds

teemingc committed:
Merge branch 'main' into unsafe_content_security_policy_fix

teemingc committed:
Clarify fix for inline styles in announcer

Updated the fix message to clarify the removal of inline styles for CSP compliance.

teemingc committed:
Merge branch 'main' into unsafe_content_security_policy_fix

teemingc committed:
Merge branch 'main' into unsafe_content_security_policy_fix

teemingc committed:
fix test

teemingc commented on packages/adapter-static/test/apps/spa/test/test.js:
```diff
@@ -46,6 +46,6 @@ test.describe(() => {
 		});
 		await page.goto('/');
-		expect(requests.length).toBe(1);
+		expect(requests.length).toBe(2);
```
our root component + user's root layout css

teemingc commented on packages/kit/src/exports/vite/build/build_server.js:
```diff
@@ -106,6 +106,16 @@ export function build_server_nodes(
 		}
 	}
 
+	/** @type {string[] | undefined} */
+	let root_stylesheets;
+	if (client_manifest && output_config.bundleStrategy === 'split') {
+		root_stylesheets = find_deps(
+			client_manifest,
+			`${normalizePath(kit.outDir)}/generated/client-optimized/app.js`,
+			false
+		).stylesheets;
+	}
```
Now eagerly loads styles from our root component

teemingc commented:
Going to let another maintainer review this because I added some code to get this working. I fixed it so that it works with AMP too because SvelteKit now "sees" the styles declared in our root component, meaning they get inlined or preloaded as required by the app

teemingc commented on .changeset/some-dryers-care.md:
```diff
@@ -0,0 +1,5 @@
+"@sveltejs/kit": patch
+---
+
+fix: remove inline styles from announcer so that the `style-src` CSP works
```
```suggestion
fix: remove inline styles from announcer
```

teemingc committed:
fix tests

teemingc committed:
fix test

teemingc committed:
Merge branch 'main' into unsafe_content_security_policy_fix

Rich-Harris commented:
Coming back to this — it works, but the extra `.css` file is a bit of a shame. What if we just modified the CSP headers when `style-src` is specified, the way we do for `script-src`?

teemingc commented:
> Coming back to this — it works, but the extra `.css` file is a bit of a shame. What if we just modified the CSP headers when `style-src` is specified, the way we do for `script-src`?

Yeah, that's a preferable solution. See https://github.com/sveltejs/kit/issues/11747#issuecomment-3800961700
